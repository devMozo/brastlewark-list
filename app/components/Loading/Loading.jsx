import React from 'react';

export const Loading = ({ color }) => {
  return <div className={`Loading Loading--${color ? color : ''} margin-2-y`} />;
};
