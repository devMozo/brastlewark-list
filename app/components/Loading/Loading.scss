.Loading {
  width: 50px;
  height: 50px;
  margin: 0 auto;
  border: 5px solid #ddd;
  border-top: 5px solid white !important;
  border-radius: 100%;
  -webkit-animation: spin 1s linear infinite;
  -moz-animation: spin 1s linear infinite;
  animation: spin 1s linear infinite;

  &.Loading--red {
    border-color: #ee5487;
  }

  &.Loading--orange {
    border-color: #f6901a;
  }

  &.Loading--whiteOrange {
    border-color: #ec9b3b;
  }

  &.Loading--green {
    border-color: #82bb30;
  }

  &.Loading--blue {
    border-color: #4facff;
  }

  @-moz-keyframes spin {
    100% {
      -moz-transform: rotate(360deg);
    }
  }
  @-webkit-keyframes spin {
    100% {
      -webkit-transform: rotate(360deg);
    }
  }
  @keyframes spin {
    100% {
      -webkit-transform: rotate(360deg);
      transform: rotate(360deg);
    }
  }
}
