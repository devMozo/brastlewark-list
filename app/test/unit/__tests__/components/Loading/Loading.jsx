import React from 'react';
import { Loading } from 'components/Loading/Loading';
import { render } from 'enzyme';

describe('<Loading />', () => {
  let props;
  let loading;

  beforeEach(() => {
    props = {
      color: undefined,
    };
    loading = () => {
      return render(<Loading {...props} />);
    };
  });

  it('Snapshotesting completed <Loading />', () => {
    props = {
      color: 'red',
    };
    expect(loading()).toMatchSnapshot();
  });

  it('Snapshotesting <Loading /> without props', () => {
    expect(loading()).toMatchSnapshot();
  });
});
