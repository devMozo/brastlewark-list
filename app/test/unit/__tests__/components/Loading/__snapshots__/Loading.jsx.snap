// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> Snapshotesting <Loading /> without props 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "Loading Loading-- margin-2-y",
    },
    "children": Array [],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`<Loading /> Snapshotesting completed <Loading /> 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "Loading Loading--red margin-2-y",
    },
    "children": Array [],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
